src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -866,6 +886,9 @@ private TexlNode ParseOperand()
                     return new SelfNode(ref _idNext, _curs.TokMove());
 
                 case TokKind.Eof:
+                    return CreateError(_curs.TokCur, TexlStrings.ErrOperandExpected);
+
+                case TokKind.Semicolon:
```

Comment:
I added another case to ParseOperand for semicolon. If we hit a semicolon token when we expect an operand to parse, we just return the error (without moving the token) and stop recursing, and continue parsing the rest of the named formula as normal.